apps/dashr-oil-and-gas/app.R

```diff
@@ -0,0 +1,829 @@
+              ),
+              value="productive",
+              labelStyle=list("display"= "inline-block"),
+              className="dcc_control"
+            ),
+            dccDropdown(
+              id="well_types",
+              options=well_type_options,
+              multi=TRUE,
+              value=list(labels(WELL_TYPES)),
+              className="dcc_control"
+            )
+          ),
+          className="pretty_container four columns",
+          id="cross-filter-options"
+        ),
+        htmlDiv(
+          list(
+            htmlDiv(
+              list(
+                htmlDiv(
+                  list(htmlH6(id="well_text"), htmlP("No. of Wells")),
+                  id="wells",
+                  className="mini_container"
+                ),
+                htmlDiv(
+                  list(htmlH6(id="gasText"), htmlP("Gas")),
+                  id="gas",
+                  className="mini_container"
+                ),
+                htmlDiv(
+                  list(htmlH6(id="oilText"), htmlP("Oil")),
+                  id="oil",
+                  className="mini_container"
+                ),
+                htmlDiv(
+                  list(htmlH6(id="waterText"), htmlP("Water")),
+                  id="water",
+                  className="mini_container"
+                )
+              ),
+              id="info-container",
+              className="row container-display"
+            ),
+            htmlDiv(
+              list(dccGraph(id="count_graph")),
+              id="countGraphContainer",
+              className="pretty_container"
+            )
+          ),
+          id="right-column",
+          className="eight columns"
+        )
+      ),
+      className="row flex-display"
+    ),
+    htmlDiv(
+      list(
+        htmlDiv(
+          list(dccGraph(id="main_graph")),
+          className="pretty_container seven columns"
+        ),
+        htmlDiv(
+          list(dccGraph(id="individual_graph")),
+          className="pretty_container five columns"
+        )
+      ),
+      className="row flex-display"
+    ),
+    htmlDiv(
+      list(
+        htmlDiv(
+          list(dccGraph(id="pie_graph")),
+          className="pretty_container seven columns"
+        ),
+        htmlDiv(
+          list(dccGraph(id="aggregate_graph")),
+          className="pretty_container five columns"
+        )
+      ),
+      className="row flex-display"
+    )
+  ),
+  id="mainContainer",
+  style=list("display"= "flex", "flex-direction"= "column")
+))
+
+filter_Dataframe <- function(df, 
+                             well_statuses, 
+                             well_types, 
+                             year_slider){
+  df <- as.data.table(df)
+  min_year <- as.Date(paste0(year_slider[[1]], "-01-01"))
+  max_year <- as.Date(paste0(year_slider[[2]], "-01-01"))
+  df <- df[df$Well_Status %in% well_statuses & df$Well_Type %in% well_types,]
```

Comment:
One nice thing about having `df` as a `data.table` is that you can avoid retyping `df` when you subset. I think that `df[Well_Status %in% well_statuses & Well_Type %in% well_types,]` would work just as well. Makes little difference here, but I find it much nicer to write without having to repeat `df$`...